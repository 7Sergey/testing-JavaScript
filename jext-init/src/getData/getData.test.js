const axios = require('axios');
const getData = require('./getData');

jest.mock('axios'); //замокали сторонние модули

describe('getData', () => {
  let response;
  beforeEach(() => {
    //чтобы для каждого теста замокать данные
    response = {
      data: [
        { id: 1, name: 'Leanne Graham' },
        { id: 2, name: 'Ervin Howell' },
        { id: 3, name: 'Clementine Bauch' },
      ],
    };
  });
  test('Корректные значения', async () => {
    axios.get.mockReturnValue(response);
    const data = await getData();
    expect(axios.get).toBeCalledTimes(1);
    expect(data).toEqual(['1', '2', '3']);
    expect(data).toMatchSnapshot();
    screen.debug();
  });
});
