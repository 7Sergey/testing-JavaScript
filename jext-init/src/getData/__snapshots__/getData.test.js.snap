// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData Корректные значения 1`] = `
[
  "1",
  "2",
  "3",
]
`;
